frontend/src/components/module-viewer/content/blocks/MatchBlock.tsx

```diff
@@ -0,0 +1,278 @@
+        </Box>
+      )}
+    </Draggable>
+  );
+};
+
+const DropAreaStyle = ({
+  children,
+}: {
+  children?: React.ReactElement;
+}): React.ReactElement => {
+  return (
+    <SimpleGrid
+      w="100%"
+      border="2px"
+      borderColor="#E5E7EB"
+      borderStyle="dashed"
+      borderRadius="4px"
+      bg="#F4F4F4"
+      columns={1}
+    >
+      <Box w="100%" gridColumnStart={1} gridRowStart={1} zIndex={2}>
+        {children}
+      </Box>
+      <Text
+        padding="1rem"
+        variant="sm"
+        color="#666"
+        gridColumnStart={1}
+        gridRowStart={1}
+      >
+        Drag & Drop Answer Here
+      </Text>
+    </SimpleGrid>
+  );
+};
+
+const MatchBlock = ({
+  block: { content },
+  editable,
+}: ContentBlockProps<MatchBlockState>): React.ReactElement => {
+  const [pairs, setPairs] = useState(Array(content.matches.length).fill(null));
+  const [isCompleted, setCompleted] = useState(false);
+
+  const setAnswer = (...args: [number | string, number | string | null][]) => {
+    const newPairs = [...pairs];
+    args.forEach(([index, newVal]) => {
+      const i = typeof index === "string" ? parseInt(index, 10) : index;
+      const val = typeof newVal === "string" ? parseInt(newVal, 10) : newVal;
+      newPairs[i] = val;
+    });
+    setPairs(newPairs);
+  };
+
+  const onDragEnd = ({ source, destination, draggableId }: DropResult) => {
+    if (destination?.droppableId === "POOL") {
+      if (source.droppableId === "POOL") return;
+      const [index] = source.droppableId.split("-");
+      setAnswer([index, null]);
+    } else if (destination?.droppableId.includes("DROP")) {
+      const [dropIndex] = destination.droppableId.split("-");
+      const [answerIndex] = draggableId.split("-");
+      // D & D between DROP locations
+      if (source.droppableId.includes("DROP")) {
+        const [sourceIndex] = source.droppableId.split("-");
+        setAnswer([sourceIndex, null], [dropIndex, answerIndex]);
+      } else {
+        setAnswer([dropIndex, answerIndex]);
+      }
+    }
+  };
+
+  const renderMatchBlock = () => {
+    return (
+      <VStack
+        w="80%"
+        padding="2rem"
+        paddingLeft="4rem"
+        paddingRight="4rem"
+        align="left"
+        borderTopColor="brand.royal"
+        borderTopWidth="10px"
+        borderRadius="8px"
+        boxShadow="xl"
+      >
+        <Flex align="center">
+          <MatchIcon />
+          <Text variant="heading">Matching</Text>
+        </Flex>
+        <Text variant="subheading">{content.question}</Text>
+        <SimpleGrid
+          templateColumns={`1fr 50% ${isCompleted ? "30px" : ""}`}
+          alignItems="center"
+          gap={2}
+          w="100%"
+        >
+          {content.matches.map(({ prompt, answer }, i) => {
+            let status = AnswerStatus.Default;
+            if (isCompleted) {
+              status =
+                pairs[i] === i ? AnswerStatus.Correct : AnswerStatus.Incorrect;

```

Comment:
I may be misunderstanding the logic here but I think doing this sort of comparison would be annoying to users in some cases. In particular, it doesn't allow admins to create multiple answers with the same text, since then users wouldn't know which corresponds to which. My suggestion: compare by strings if this check fails.